src/main/java/mate/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,45 @@
+package mate.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private static final String DRIVER_ID = "driver_id";
+    private Set<String> allowedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrls = new HashSet<>();
+        allowedUrls.add("/login");
+        allowedUrls.add("/index");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        HttpSession session = req.getSession();
+        Long driverId = (Long) session.getAttribute(DRIVER_ID);
+        if (driverId == null && allowedUrls.contains(req.getServletPath())) {
```

Comment:
do we need this check here "driverId == null" ?

Reply:
no